### PR TITLE
Added Windvane ADC Support

### PR DESCRIPTION
**Pull request by computergeek1507**
## Description:
Added support for Windvane direction based on ADC value.
Tested with the [Sparkfun Weather Kit ](https://www.sparkfun.com/products/15901) and a Wemos D1 mini.

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.10
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_xsns_sensor/xsns_02_analog.ino [hunk 1]
```diff
@@ -707,6 +710,48 @@ void AdcShow(bool json) {
...
+        }
+        if(dir_val > 2.00 && dir_val < 2.20 ) {
+          wind_chr = D_TX20_SOUTH D_TX20_WEST;
+        }
+        if(dir_val > 3.00 && dir_val < 3.25 ) {
+          wind_chr = D_TX20_WEST;
+        }
+        if(dir_val > 2.8 && dir_val < 2.99 ) {
+          wind_chr = D_TX20_NORTH D_TX20_WEST;
+        }
```
Comment (barbudor) [inline tasmota/tasmota_xsns_sensor/xsns_02_analog.ino:743]: Number of tests can be halved in 2 by ordering properly and using else if
Also this leaves empty holes like 2.995 which falls in no cases

Just an opinion: I know you didn't invented those direction macros but wouldn't an angle easier to process downstream in a JSON data ?
I could make sense to use human readable directions in hte Web UI but angle in the JSON

Comment (sfromis) [reply tasmota/tasmota_xsns_sensor/xsns_02_analog.ino:743]: I agree that an angle would be preferable for JSON, and the web UI is not meant to be "the useful place" anyway.

And it does look odd that values 
0.75..1.20 is south
1.30..1.70 is northeast
2.00..2.20 is southwest
2.50..2.70 is north
I was expecting the analog value to have a circular dependence on the angle, but your values are jumping around. Is that really correct?

Comment (computergeek1507) [reply tasmota/tasmota_xsns_sensor/xsns_02_analog.ino:743]: It uses a resistor network with reed switches according to the data sheet, https://cdn.sparkfun.com/assets/d/1/e/0/6/DS-15901-Weather_Meter.pdf

File: tasmota/include/tasmota_template.h [hunk 2]
```diff
@@ -181,6 +181,7 @@ enum UserSelectablePins {
   GPIO_OPTION_E,                       // Emulated module
   GPIO_SDM230_TX, GPIO_SDM230_RX,      // SDM230 Serial interface
   GPIO_ADC_MQ,                         // Analog MQ Sensor
+  GPIO_ADC_WIND_DIR,                         // Analog Wind Direction
```
Comment (arendst) [inline tasmota/include/tasmota_template.h:184]: This needs to move to the end of the list

File: tasmota/include/tasmota_template.h [hunk 3]
```diff
@@ -434,6 +435,7 @@ const char kSensorNames[] PROGMEM =
   D_SENSOR_OPTION " E|"
   D_SENSOR_SDM230_TX "|" D_SENSOR_SDM230_RX "|"
   D_SENSOR_ADC_MQ "|"
+  D_SENSOR_ADC_WIND_DIR  "|"
```
Comment (arendst) [inline tasmota/include/tasmota_template.h:438]: Ditto.

File: tasmota/tasmota_xsns_sensor/xsns_02_analog.ino (hunk 1 above)
Comment (arendst) [reply tasmota/tasmota_xsns_sensor/xsns_02_analog.ino:743]: I suggest to convert the received values to degrees and use that to index in the winddirection text string like TX20 does but then without the NNE options. Then output both string and degrees again like TX20 does.

Comment (arendst) [reply tasmota/tasmota_xsns_sensor/xsns_02_analog.ino:743]: In addition move from ADC float ASAP to degree integer lowering code size considerably.

**Comment by Jason2866** [comment]
@computergeek1507 Are you going to change the code?

**Pull request closed without merging**